cmd/kubeadm/app/cmd/phases/join/controlplanejoin.go

```diff
@@ -0,0 +1,256 @@
+			name:  "controller-manager",
+			short: getPhaseDescription(kubeadmconstants.KubeControllerManager),
+		},
+		kubeadmconstants.KubeScheduler: {
+			name:  "scheduler",
+			short: getPhaseDescription(kubeadmconstants.KubeScheduler),
+		},
+	}
+)
+
+type controlPlaneJoinData interface {
+	Cfg() *kubeadmapi.JoinConfiguration
+	ClientSetFromFile(string) (*clientset.Clientset, error)
+	InitCfg() (*kubeadmapi.InitConfiguration, error)
+}
+
+func getPhaseDescription(component string) string {
+	return fmt.Sprintf("Generates the %s static Pod manifest", component)
+}
+
+func getControlPlaneJoinPhaseFlags() []string {
+	return []string{
+		options.APIServerAdvertiseAddress,
+		options.APIServerBindPort,
+		options.CfgPath,
+		options.ControlPlane,
+		options.NodeName,
+		options.TokenDiscovery,
+		options.TokenDiscoveryCAHash,
+		options.TokenDiscoverySkipCAHash,
+	}
+}
+
+// NewControlPlaneJoinPhase creates a kubeadm workflow phase that implements marking the new node as master and update the cluster status
+// with information about current node
+func NewControlPlaneJoinPhase() workflow.Phase {
+	return workflow.Phase{
+		Name:  "control-plane-join",
+		Short: "Prepares the machine to join a control plane.",
+		Long:  cmdutil.MacroCommandLongDescription,
+		Phases: []workflow.Phase{
+			{
+				Name:           "all",
+				Short:          "Generates all static Pod manifest files",
+				InheritFlags:   getControlPlaneJoinPhaseFlags(),
+				RunAllSiblings: true,
+			},
+			newControlPlaneJoinSubphase(kubeadmconstants.KubeAPIServer),
+			newControlPlaneJoinSubphase(kubeadmconstants.KubeControllerManager),
+			newControlPlaneJoinSubphase(kubeadmconstants.KubeScheduler),
+			newEtcdLocalSubphase(),
+			newUploadConfigSubphase(),
+			newMarkControlPlaneSubphases(),
+		},
+		InheritFlags: getControlPlaneJoinPhaseFlags(),
+	}
+}
+
+func newControlPlaneJoinSubphase(component string) workflow.Phase {
+	return workflow.Phase{
+		Name:         controlPlanePhaseProperties[component].name,
+		Short:        controlPlanePhaseProperties[component].short,
+		Run:          runControlPlaneJoinSubphase(component),
+		InheritFlags: getControlPlaneJoinPhaseFlags(),
+	}
+}
+
+func newEtcdLocalSubphase() workflow.Phase {
+	return workflow.Phase{
+		Name:         "local",
+		Short:        "Generates the static Pod manifest file for a local etcd member",
+		Run:          runEtcdLocalPhase,
+		InheritFlags: getControlPlaneJoinPhaseFlags(),
+	}
+}
+
+func newUploadConfigSubphase() workflow.Phase {
+	return workflow.Phase{
+		Name:         "upload-config",
+		Short:        "Upload the currently used configuration to the cluster",
+		Run:          runUploadConfigPhaseLocal,
+		InheritFlags: getControlPlaneJoinPhaseFlags(),
+	}
+}
+
+func newMarkControlPlaneSubphases() workflow.Phase {
+	return workflow.Phase{
+		Name:         "mark-control-plane",
+		Short:        "Mark a node as a control-plane",
+		Run:          runMarkControlPlanePhaseLocal,
```

Comment:
try removing `Local` from the function name.

Reply:
done